src/base/utils/io.h

```diff
@@ -98,8 +98,7 @@ namespace Utils::IO
     };
 
     // TODO: define a specific type for `additionalMode`
-    // providing `size` is explicit and is strongly recommended
-    nonstd::expected<QByteArray, ReadError> readFile(const Path &path, qint64 size, QIODevice::OpenMode additionalMode = {});
+    nonstd::expected<QByteArray, ReadError> readFile(const Path &path, qint64 size = -1, QIODevice::OpenMode additionalMode = {});
```

Comment:
Why change it?
I would not wish to give the impression that `size` is an optional parameter. This is against the original intention of creating this function and even the comments were clear about it...

Reply:
>But it IS essentially optional feature of "reading file" task.

I would say how an interface is designed is all about intention. IMO conventions comes later than intentions. C++ is full of intentions...

>Otherwise, you impose the idea of "inventing" some kind of limits on someone, even in cases where it is unnatural.

In this case, it is good to be explicit for coders to think about it IMO.

Also note that, since I don't really have enough time to participate in reviews, I don't intend to raise a blocker. But at the same time I feel this change deserved its own commit and explanation rather than included in some other commit.

Reply:
> In this case, it is good to be explicit for coders to think about it IMO.
> 
> Also note that, since I don't really have enough time to participate in reviews, I don't intend to raise a blocker. But at the same time I feel this change deserved its own commit and explanation rather than included in some other commit.

OK, reverted.